drivers/cache/Kconfig.kinetis

```diff
@@ -0,0 +1,16 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright 2022 NXP
+
+config CACHE_KINETIS
+bool "Kinetis cache driver"
+	select CACHE_HAS_DRIVER
+	select CACHE_MANAGEMENT
+	help
+	  This option enables the CACHE driver for NXP kinetis caches.
+
+if CACHE_KINETIS
+choice CACHE_TYPE
+	default HAS_EXTERNAL_CACHE
+endchoice
```

Comment:
This should be in the SoC / board Kconfig.

Reply:
Why. It would just have to be in multiple files instead of one then. The situation is that some NXP SOC's have a custom cache. Any of them that do should select CACHE_KINETIS in their soc file. I don't see a reason to include 4 more lines in every SOC file for soc's with this cache rather than just have the 4 lines in one place.

Reply:
ok, leave it here. Can you just have:
```
choice CACHE_TYPE
	default HAS_EXTERNAL_CACHE if CACHE_KINESIS
```